clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -810,89 +854,87 @@ public void testGroupMetadataAfterCreationWithGroupIdIsNotNullAndGroupInstanceId
         final Properties props = requiredConsumerPropertiesAndGroupId(groupId);
         props.put(ConsumerConfig.GROUP_INSTANCE_ID_CONFIG, groupInstanceId);
         final ConsumerConfig config = new ConsumerConfig(props);
-        try (final AsyncKafkaConsumer<String, String> consumer =
-            new AsyncKafkaConsumer<>(config, new StringDeserializer(), new StringDeserializer())) {
-
-            final ConsumerGroupMetadata groupMetadata = consumer.groupMetadata();
-
-            assertEquals(groupId, groupMetadata.groupId());
-            assertEquals(Optional.of(groupInstanceId), groupMetadata.groupInstanceId());
-            assertEquals(JoinGroupRequest.UNKNOWN_GENERATION_ID, groupMetadata.generationId());
-            assertEquals(JoinGroupRequest.UNKNOWN_MEMBER_ID, groupMetadata.memberId());
-        }
+        final AsyncKafkaConsumer<String, String> consumer =
+            new AsyncKafkaConsumer<>(config, new StringDeserializer(), new StringDeserializer());
+        final ConsumerGroupMetadata groupMetadata = consumer.groupMetadata();
+        assertEquals(groupId, groupMetadata.groupId());
+        assertEquals(Optional.of(groupInstanceId), groupMetadata.groupInstanceId());
+        assertEquals(JoinGroupRequest.UNKNOWN_GENERATION_ID, groupMetadata.generationId());
+        assertEquals(JoinGroupRequest.UNKNOWN_MEMBER_ID, groupMetadata.memberId());
+        consumer.close(Duration.ZERO);
     }
 
     @Test
     public void testGroupMetadataUpdateSingleCall() {
         final String groupId = "consumerGroupA";
         final ConsumerConfig config = new ConsumerConfig(requiredConsumerPropertiesAndGroupId(groupId));
         final LinkedBlockingQueue<BackgroundEvent> backgroundEventQueue = new LinkedBlockingQueue<>();
-        try (final AsyncKafkaConsumer<String, String> consumer =
```

Comment:
Please be aware that all these changes will be unnecessary/conflicting once we move to mocks, which is rather soon I believe, see:

https://github.com/apache/kafka/pull/14930

We will get lots of conflicts and I think whoever has to resolve the conflicts will probably just skip the changes from this PR, because the commit logic in the background thread will not be executed anymore in this test.

Reply:
ok makes sense, I can revert these changes.

Reply:
I didn't mean to revert the changes, but discard them when we merge with the asyncKafkaConsumer refactoring. Is it possible these changes were required to not run OOM?

One option would be to rebase this PR on the refactoring, which can possibly resolve these OOMs.